libevmasm/Instruction.cpp

```diff
@@ -326,12 +327,9 @@ void dev::solidity::eachInstruction(
 		if (isValidInstruction(instr))
 			additional = instructionInfo(instr).additional;
 		u256 data;
-		for (size_t i = 0; i < additional; ++i)
-		{
-			data <<= 8;
-			if (++it < _mem.end())
-				data |= *it;
-		}
+		for (size_t i = 0; i < std::min(additional, static_cast<size_t>(std::distance(it, _mem.end())) - 1); ++i, --additional)
+			data = (data << 8) + *++it;
+		data <<= 8 * additional; // pad the remaining number of additional octets with zeros
 		_onInstruction(instr, data);
```

Comment:
I think the "lazy" solution of adding a ``if (it == _mem.end()) break;`` here would actually be more readable in the end.